R/summarize_weighted_production.R

```diff
@@ -211,6 +211,18 @@ add_technology_share <- function(data) {
     group_by(!!!dplyr::groups(data))
 }
 
+add_technology_share_target <- function(data) {
+  crucial <- c("production_target", "sector_ald", "year", "technology")
+
+  check_crucial_names(data, crucial)
+  walk_(crucial, ~ check_no_value_is_missing(data, .x))
+
+  data %>%
+    group_by(.data$sector_ald, .data$year, .data$scenario, .data$name_ald) %>%
+    mutate(technology_share_target = .data$production_target / sum(.data$production_target)) %>%
```

Comment:
Do we also need to check for crucial names "scenario" and "name_ald"? 

It would be more readable if `crucial` holds the names in the order they show up in the code, or in alphabetical order so they are easier to lookup.

Reply:
I'm actually not sure what the protocol is here. These are internal only functions, and used in dplyr chains on data that has already had it's crucial columns checked. 

Would it make sense to remove the crucial check altogether?

Reply:
Oh sure! You are right!  I think it's best to check names only once at the top level.